src/logic/basic.lean

```diff
@@ -875,6 +875,10 @@ lemma heq_of_cast_eq :
   ∀ {α β : Sort*} {a : α} {a' : β} (e : α = β) (h₂ : cast e a = a'), a == a'
 | α ._ a a' rfl h := eq.rec_on h (heq.refl _)
 
+lemma congr_fun_heq {α β γ : Sort*} {f : α → γ} {g : β → γ} (h₁ : α = β) (h₂ : f == g) (x : α) :
+  g (cast h₁ x) = f x :=
```

Comment:
My only concern is about switching the order of `f` and `g` between the hypotheses and the conclusion. Wouldn't it be better to be consistent with `congr_fun`?

Reply:
I wrote it down this way since the general rule is that complicated things go on the left, but I agree that consistency with `congr_fun` overrides this.

Reply:
It would be possible to have it both ways, and start with a `y : \b`, and then cast that before feeding it to `f`. But this is fine as is.

Reply:
I like that approach.

Reply:
Wait, I just realized something. By having `cast h₁.symm x` instead of `cast h₁ x`, the argument `h₁` can no longer be inferred. The fix would be taking in an argument `h₁ : β = α`. What do you think about that?